Better MUC light room config (#2536)
esl/MongooseIM: doc/migrations/3.5.0_3.5.0plus.md

## Different MUC Light room schema definition

> Separate MUC Light options for the schema and default values were somewhat unintuitive and led to problems with RDBMS backend.

Reviewer: We have introduced a change that enforces defining fields with default values.
The previous setup led to problems with the RDBMS backend as separating MUC Light options for the schema from the default values was unintuitive.
In a specific case when the default config was a subset of the schema and the client failed to provide these values when a room was created, it stored the incomplete config in the table. Then the missing config fields could not be supplied by the clients.
If you've experienced this issue, a way to fix it is described in the [Known issues](../operation-and-maintenance/known-issues.md) page.

                  {"subject", "Chit-chat"}
                 ]}
```

**New config:**

```
{config_schema, [
                 {"roomname", "The room"},
                 {"subject", "Chit-chat"},
                 {"background", ""},
                 {"notification_sound", ""}
                ]}
```

#### Example 2

**Old config:**

```
{config_schema, [
                 "roomname",
                 {"subject", binary},
                 {"priority", priority, integer},
                 {"owners-height", owners_height, float}
                ]},
{default_config, [
                  {"roomname", "The room"},
                  {"subject", "Chit-chat"},
                  {"priority", 10}]}
```

**New config:**

```
{config_schema, [
                 {"roomname", "The room"},
                 {"subject", "Chit-chat"},
                 {"priority", 10, priority, integer},
                 {"owners-height", 180.0, owners_height, float}
                ]}
                 
```

### What has changed? - for developers

The room config schema is currently stored in a completely different data structure, so if you have any custom modules that use it, you'll need to adjust them.

> What is more, all definitions and the room config API have been extracted from `mod_muc_light.hrl` and `mod_muc_light_utils.erl` into `mod_muc_light_room_config.erl` module.

Reviewer: ```suggestion
Additionally,  all definitions and the room config API have been extracted from `mod_muc_light.hrl` and `mod_muc_light_utils.erl` into `mod_muc_light_room_config.erl` module.
```

> For more information, please check the types and functions specs in the aforementioned file.

Reviewer: ```suggestion
For more information, please check the specs for types and functions in the aforementioned file.
```
